File: backend/src/main/java/turip/creator/service/CreatorService.java
package turip.creator.service;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import turip.creator.controller.dto.response.CreatorResponse;
import turip.creator.domain.Creator;
import turip.creator.repository.CreatorRepository;
import turip.exception.NotFoundException;

@Service
@RequiredArgsConstructor
public class CreatorService {

    private final CreatorRepository creatorRepository;

    public CreatorResponse getById(Long id) {
        Creator creator = creatorRepository.findById(id)
                .orElseThrow(() -> new NotFoundException("크리에이터를 찾을 수 없습니다."));
        return CreatorResponse.from(creator);
    }
}

File: android/app/src/main/java/com/on/turip/data/creator/service/CreatorService.kt
package com.on.turip.data.creator.service

import com.on.turip.data.creator.dto.CreatorResponse
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

interface CreatorService {
    @GET("creators/{creatorId}")
    suspend fun getCreator(
        @Path("creatorId") creatorId: Long,
    ): Response<CreatorResponse>
}
